docs/extensions.md

```diff
@@ -0,0 +1,697 @@
+# Writing PHP Extensions in Go
+
+With FrankenPHP, you can **write PHP extensions in Go**, which allows you to create **high-performance native functions** that can be called directly from PHP. Your applications can leverage any existing or new Go library, as well as the infamous concurrency model of **goroutines right from your PHP code**.
+
+Writing PHP extensions is typically done in C, but it's also possible to write them in other languages with a bit of extra work. PHP extensions allow you to leverage the power of low-level languages to extend PHP's functionalities, for example, by adding native functions or optimizing specific operations.
+
+Thanks to Caddy modules, you can write PHP extensions in Go and integrate them very quickly into FrankenPHP.
+
+## Two Approaches
+
+FrankenPHP provides two ways to create PHP extensions in Go:
+
+1. **Manual Implementation** - Full control over the extension structure
+2. **Using the Extension Generator** - A simplified approach that generates all necessary boilerplate
+
+We'll start with the manual approach to understand how extensions work under the hood, then show how the generator simplifies the process.
+
+## Manual Implementation
```

Comment:
I would start with the easiest option (code generator), then show the advanced way.

Reply:
PR updated